rfcs/2021-01-05-typography-scaling.md

```diff
@@ -0,0 +1,200 @@
+
+Considering blade we generally have a simple formula that derives the line-height and we consider it to be 150% of the font-size. i.e,
+
+>**`line-height = font-size * 1.5`**
+
+And will round-off the resulting value.
+
+### **Current Limitations**
+There are two major limitations while scaling typography to multiple breakpoints,
+1. **It does not cater to landing/marketing pages.** Which are generally much larger compared to the size-ratio on the dashboard (Maybe a [Major Third](https://type-scale.com/?size=14&scale=1.125&text=A%20Visual%20Type%20Scale&font=Lato&fontweight=400&bodyfont=Poppins&bodyfontweight=400&lineheight=1.65&backgroundcolor=white&fontcolor=%23333&preview=false) scale will work better). This is currently being looked at by the comm. design team based on the same principles.
+2. **Same type scaling across all devices/breakpoints.** After the type scale is defined, we also need to define the scale across multiple device sizes. There are few approaches that we've came up with and after research from other DS/products like *GitHub*, *Eightshapes*, *Paystack*, *Stripe*, and a few more we've observed that for dashboards generally tweaking the **heading** sizes works best and the base font-size will remain same for all other devices/breakpoints.
+
+We'll look into the responsive type scale and see what approach fits best for our use case.
+
+>📝 Note: **Responsive Typography** will scale in segments. Which means if `h1` at `1440px` is `40px`, then at `991px` it will become `32px` directly rather than going from `40px -> 38px -> 36px -> 34px -> 32px...`.
+
+### **Our Approach**
+>**Similar vertical & horizontal ratio across multiple breakpoints (for headings)**
+
+Our product is majorly used on mobiles/tablets/desktops. Therefore, We'll  consider only two breakpoints for the typography scale, one for mobiles & one for both desktops & tablets. Which is for devices having width `>=991px` (approx.)
+
+![Scaling: Similar vertical & horizontal ratio](./images/typography/typography-same-vertical-horizontal-ratio.png)
+
+In this approach, we use the same ratio of `1.125` down across each breakpoint. Pretty straightforward, we multiply everything down and across by the same ratio.
+
+### **Example:**
+| Style  | Mobile | Desktops/Tablets |
+|--------|--------|------------------|
+| `<h6>` | 14px   | 16px             |
+| `<h5>` | 16px   | 18px             |
+| `<h4>` | 18px   | 20px             |
+| `<h3>` | 20px   | 25px             |
+| `<h2>` | 25px   | 32px             |
+| `<h1>` | 32px   | 40px             |
+
+That means, for devices like, **desktops/tablets** if font-size for `<h6> = 16px` than, for **mobiles** font-size for `<h6> = (16 / 1.125) = 14px`.
+
+>📝 Note: Here we're rounding-off all the numbers for the sake of simplicity and to align the content according to the grid-system. Also, various browsers treat fractional font sizes differently.
+
+This is how it will look like across desktop & mobile.
```

Comment:
maybe add annotations to show the font sizes? This screenshot just shows the headings